parlai/core/torch_generator_agent.py

```diff
@@ -1123,7 +1124,10 @@ def advance(self, logprobs):
         #  check new hypos for eos label, if we have some, add to finished
         for hypid in range(self.beam_size):
             if self.outputs[-1][hypid] == self.eos:
-                if self.scores[hypid] == neginf(self.scores.dtype):
+                if (
+                    self.scores[hypid] == neginf(self.scores.dtype)
+                    or self.scores[hypid].item() == -math.inf
+                ):
```

Comment:
Let me know if this breaks any assumptions. But it seems like neither a (output, score) progression of `[(ok, -4.3906), (eos, -1.9766), (eos, -65504.0]` nor `[(ok, -4.3906), (eos, -1.9766), (eos, -inf)]` should be added to the stack of finished hypotheses.

Reply:
Maybe we should just do < instead of ==? With your clamping, it's not obvious to me that that the == -inf can even be reached, but maybe we should just catch anything crazier than inf.

Reply:
Good idea, I think `<` makes more sense.